packages/rpc-core/src/rpc-methods/getSignaturesForAddress.ts

```diff
@@ -0,0 +1,44 @@
+import { Base58EncodedAddress } from '@solana/keys';
+import { Commitment, RpcResponse, Slot, TransactionSignature, U64UnsafeBeyond2Pow53Minus1 } from './common';
+
+type GetSignaturesForAddressTransaction = RpcResponse<{
+    /** transaction signature as base-58 encoded string */
+    signature: TransactionSignature;
+    /** The slot that contains the block with the transaction */
+    slot: Slot;
+    /** Error if transaction failed, null if transaction succeeded. */
+    err: object | null;
+    /** Memo associated with the transaction, null if no memo is present */
+    memo: string | null;
+    /** estimated production time, as Unix timestamp (seconds since the Unix epoch) of when transaction was processed. null if not available. */
+    blockTime: U64UnsafeBeyond2Pow53Minus1 | null;
+    /** The transaction's cluster confirmation status */
+    confirmationStatus: Commitment | null;
+}>;
+
+type GetSignaturesForAddressApiResponse = [GetSignaturesForAddressTransaction];
+
+export type AllowedCommitmentForGetSignaturesForAddress = Exclude<Commitment, 'processed'>;
+
+type GetSignaturesForAddressConfig = Readonly<{
+    commitment?: AllowedCommitmentForGetSignaturesForAddress;
+    /** The minimum slot that the request can be evaluated at */
+    minContextSlot?: Slot;
+    /** maximum transaction signatures to return (between 1 and 1,000). Default: 1000 */
+    limit?: number;
+    /** start searching backwards from this transaction signature. If not provided the search starts from the top of the highest max confirmed block. */
+    before?: TransactionSignature;
+    /** search until this transaction signature, if found before limit reached */
+    until?: TransactionSignature;
```

Comment:
Presumably you can supply `until` _without_ `before`? If you _can't_ then just turn this into multiple overloads, each enforcing a valid combination.

Reply:
Yep, I can't see any restrictions on these combinations that we can capture in these types. Neither, just `until`, just `before`, and both are all valid.